src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -258,13 +262,38 @@ func (s *service) Bootstrapped(ctx thrift.Context) (*rpc.NodeBootstrappedResult_
 }
 
 func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryResult_, error) {
+	var (
+		ctx context.Context
+		sp  opentracing.Span
+		ok  bool
+	)
+
+	ctx = tchannelthrift.Context(tctx)
+	ctx, sp, ok = ctx.StartTraceSpan(tracepoint.QuerySpanID)
```

Comment:
why do we still need the `ok`? considering you're giving back a NoopSpan when !ok - can you just avoid the `if check` and always do the `sp.LogFields(...)` jazz?

Reply:
Hey so we actually want to know ahead of time so we don't call `sp.LogFields(...)` etc in the hot path (because a lot of the time `blah.String()` or other things done during the call to `LogFields()` cause allocations, even the call itself since it is varargs allocates a slice each time).

Reply:
Hey fair enough. I tried out `sp.LogFields(...)` with 8 fields logged, it's (~300ns; 500B alloc'd) per call (https://play.golang.org/p/32Kdr-Ha4Ep). Even if we have 100 of spans created per RPC call, that's ~30us. It won't be a factor in most rpc paths (indexing/writes) but could be more significant if we were to instrument all the 10K ID fetches of a FetchBatch call. 

Would you be opposed to keeping two methods: one which does the instrumentation by default (i.e. no `ok` returned) and one which does? Can use the appropriate method depending on the call site.

Reply:
+1 -- would be great to avoid boilerplate where possible (though ofc not at the expense of perf where perf is needed)

Reply:
Sure, having two methods makes sense. 

Maybe `StartTraceSpan(...)` and `StartSampledTraceSpan(...)` where the second one is the one that returns the `bool` at the end?